frontend/src/component/admin/roles/RoleForm/RolePermissionCategories/RolePermissionProject.tsx

```diff
@@ -0,0 +1,148 @@
+import { useMemo } from 'react';
+import {
+    Box,
+    Checkbox,
+    FormControlLabel,
+    styled,
+    Typography,
+} from '@mui/material';
+import type { ICheckedPermissions, IPermission } from 'interfaces/permissions';
+import { getRoleKey } from 'utils/permissions';
+import {
+    type PermissionCategory,
+    PROJECT_PERMISSIONS_STRUCTURE,
+} from '@server/types/permissions';
+
+interface IEnvironmentPermissionAccordionProps {
+    permissions: IPermission[];
+    checkedPermissions: ICheckedPermissions;
+    onPermissionChange: (permission: IPermission) => void;
+}
+
+const StyledGrid = styled('div')(({ theme }) => ({
+    display: 'grid',
+    gridTemplateColumns: '1fr 1fr',
+    gap: theme.spacing(1),
+    [theme.breakpoints.down('md')]: {
+        gridTemplateColumns: '1fr',
+    },
+}));
+
+const StyledLabel = styled(Typography)(({ theme }) => ({
+    lineHeight: 1.2,
+    marginBottom: theme.spacing(1),
+}));
+
+const StyledSectionTitle = styled(Typography)(({ theme }) => ({
+    fontWeight: theme.typography.fontWeightBold,
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(1),
+}));
+
+export const ProjectRolePermissionCategory = ({
+    permissions,
+    checkedPermissions,
+    onPermissionChange,
+}: IEnvironmentPermissionAccordionProps) => {
+    const allPermissions = useMemo(
+        () =>
+            permissions?.map((permission) => getRoleKey(permission)).sort() ||
+            [],
+        [permissions],
+    );
+
+    const otherPermissions = useMemo(() => {
```

Comment:
I see a lot of mapping and filtering making up for the UI logic. I'd expect some tests for that